ensips/ccip-write/ccip-write.md

```diff
@@ -0,0 +1,342 @@
+
+---
+
+> ensip: <TBD>
+> 
+> 
+> title: Wildcard Writing
+> description: A standardized implementation for managing offchain domains using an External Resolver
+> author: Lucas Picollo (@pikonha), Alex T. Netto (@alextnetto)
+> discussions-to: <URL>
+> status: Draft
+> type: Standards Track
+> category: ERC
+> created: 2024-08-14
+> requires: 5559
+> 
+
+## Abstract
+
+This ENSIP proposes a standardized Offchain Resolver interface for managing offchain domains within the Ethereum Name Service (ENS) ecosystem. It addresses the growing trend of storing domains off the Ethereum blockchain to reduce transaction fees while maintaining compatibility with existing ENS components. The proposal outlines methods for domain registration, transferring, and setting records, ensuring a consistent approach to offchain domain management.
+
+## Motivation
+
+With the acceptance of CCIP-Read by the Ethereum community, there has been a notable shift towards storing domains in locations other than the Ethereum blockchain to avoid high transaction fees. This shift has revealed a significant gap: the lack of standardized methods for managing offchain domains. By establishing a standardized offchain resolver implementation and user flow, we can ensure a consistent approach to offchain domain management, enabling applications that support this EIP flow to integrate this feature and enhance user experience seamlessly, increasing scalability and providing cost-effective solutions.
+
+## Specification
+
+### Database Implementation Considerations
+
+All supported offchain writing calls MUST rely on EIP-5559, specifying the custom error `StorageHandledByOffChainDatabase`. This error MUST return all required values to implement the EIP-712 signature, ensuring domain ownership.
+
+### L2 Implementation Considerations
+
+Following EIP-5559, this specification MUST support implementation on Layer 2 solutions. The specification describes the writing strategy for L2 relying on the custom error `StorageHandledByL2` with the following arguments:
+
+1. **Chain ID**: the chain ID of the L2 network where the contract is deployed.
+2. **Contract Address**: the address of the contract on the L2 network.
+
+### Subdomain registering
+
+**Register parameters**
+
+As the initial step in registering an offchain subdomain, the `registerParams` function has been implemented to support a wide variety of use cases. This function plays a crucial role in creating a flexible and extensible offchain subdomain registration system.
+
+Given that this step relies on the CCIP-Read standard, this interface SHALL only be implemented by the resolver deployed to Ethereum since it won't be gathering the data through the function call on layer 2, but by directly accessing the storage layout of the given contract.
+
+The function has the following signature:
+
+```solidity
+struct RegisterParams {
+    uint256 price;
+    bytes extraData;
+}
+
+function registerParams(
+    bytes memory name,
```

Comment:
```suggestion
    bytes memory name,
    uint256 commitTime,
```

If commitTime == 0 then there is no commit reveal. If there is commit then it's the keccak256(registration calldata)